emerlin2caom2/main_app.py

```diff
@@ -135,34 +179,83 @@ def measurement_set_metadata(self, observation, ms_dir, pickle_obj):
 
         return plane
         ### These components need their output value to be changed somewhat
-        # provenance.inputs = pickle_obj['inputs']['fits_path']
+        # provenance.inputs = pickle_obj['fits_path']
         # provenance.keywords = str([key for key, value in pickle_obj['input_steps'].items() if value == 1])
 
+
+    def build_simple_observation(self, casa_info, pickle_dict, ante_id):
+
+        obs_id = self.basename(self.storage_name)
+
+        observation = SimpleObservation('EMERLIN', '{}_{}'.format(obs_id, ante_id))
+        observation.obs_type = 'science'
+        observation.intent = ObservationIntentType.SCIENCE
+
+
+        target_name = pickle_dict['targets']#
+        # target_pos = pickle_dict[target_name] needs updating
+        observation.target = Target('TBD')
+        observation.target.name = target_name
+        # observation.targetposition = TargetPosition()
+        # observation.targetposition = target_pos
+
+        observation.telescope = Telescope(casa_info['tel_name'][0])
+        observation.proposal = Proposal(casa_info['prop_id'])
+        if ante_id != 'lovell':
+            cart_coords = polar2cart(casa_info['ante_pos'][ante_id]['m0']['value'],
+                                     casa_info['ante_pos'][ante_id]['m1']['value'],
+                                     casa_info['ante_pos'][ante_id]['m2']['value'])
+
+            instrument_name = casa_info['antennas'][ante_id]
+        else:
+            cart_coords = polar2cart(casa_info['obs_pos']['m0']['value'],
+                                     casa_info['obs_pos']['m1']['value'],
+                                     casa_info['obs_pos']['m2']['value'])
+            instrument_name = 'lv'
+
+        observation.telescope.geo_location_x = cart_coords['x']
+        observation.telescope.geo_location_y = cart_coords['y']
+        observation.telescope.geo_location_z = cart_coords['z']
+        observation.instrument = Instrument(instrument_name)
+
+        xml_output_name = self.xml_out_dir + obs_id + '_' + str(ante_id) + '.xml'
```

Comment:
xml_output_name looks like this for the Simple Observations:
TS8004_C_001_20190801_0.xml (and onward through integer 5)
Plus this one, even though Lovell was not in the array for this Derived Observation:
TS8004_C_001_20190801_lovell.xml
Is the integer assignment part of eMERLIN protocol, or would it be possible to use the meaningful names for the Simple Observation IDs?  The code I added for ordering and selecting the first antenna for geolocs in the recent update to my date_elements branch may be useful.  Also, Lovell would be in the antenna list if it is part of the array, so we don't have to treat it as a special case.  The antenna IDs used in other eMERLIN code are: 'Lo', 'Mk2', 'Pi', 'Da', 'Kn', 'De', 'Cm' whereas in main_app.py, I see both 'lv' and 'lovell'.  Elsewhere, in creating a Simple Observation object, 'Lovell' is used.

Reply:
This should now be in, I also added the change to the obs_id as it would throw an error if these did not match when uploading